Content.Shared/Stacks/StackPrototype.cs

```diff
@@ -2,40 +2,46 @@
 using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
 using Robust.Shared.Utility;
 
-namespace Content.Shared.Stacks
+namespace Content.Shared.Stacks;
+
+[Prototype("stack")]
+public sealed class StackPrototype : IPrototype
 {
-    [Prototype("stack")]
-    public sealed class StackPrototype : IPrototype
-    {
-        [ViewVariables]
-        [IdDataField]
-        public string ID { get; } = default!;
-
-        /// <summary>
-        ///     Human-readable name for this stack type e.g. "Steel"
-        /// </summary>
-        /// <remarks>This is a localization string ID.</remarks>
-        [DataField("name")]
-        public string Name { get; } = string.Empty;
-
-        /// <summary>
-        ///     An icon that will be used to represent this stack type.
-        /// </summary>
-        [DataField("icon")]
-        public SpriteSpecifier? Icon { get; }
-
-        /// <summary>
-        ///     The entity id that will be spawned by default from this stack.
-        /// </summary>
-        [DataField("spawn", required: true, customTypeSerializer:typeof(PrototypeIdSerializer<EntityPrototype>))]
-        public string Spawn { get; } = string.Empty;
-
-        /// <summary>
-        ///     The maximum amount of things that can be in a stack.
-        ///     Can be overriden on <see cref="StackComponent"/>
-        ///     if null, simply has unlimited max count.
-        /// </summary>
-        [DataField("maxCount")]
-        public int? MaxCount { get; }
-    }
+    [ViewVariables]
+    [IdDataField]
+    public string ID { get; } = default!;
+
+    /// <summary>
+    ///     Human-readable name for this stack type e.g. "Steel"
+    /// </summary>
+    /// <remarks>This is a localization string ID.</remarks>
+    [DataField("name")]
+    public string Name { get; } = string.Empty;
+
+    /// <summary>
+    ///     An icon that will be used to represent this stack type.
+    /// </summary>
+    [DataField("icon")]
+    public SpriteSpecifier? Icon { get; }
+
+    /// <summary>
+    ///     The entity id that will be spawned by default from this stack.
+    /// </summary>
+    [DataField("spawn", required: true, customTypeSerializer:typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string Spawn { get; } = string.Empty;
+
+    /// <summary>
+    ///     The maximum amount of things that can be in a stack.
+    ///     Can be overriden on <see cref="StackComponent"/>
+    ///     if null, simply has unlimited max count.
+    /// </summary>
+    [DataField("maxCount")]
+    public int? MaxCount { get; }
+
+    /// <summary>
+    /// The size of an individual unit of this stack.
+    /// </summary>
+    [DataField("itemSize")]
+    public int? ItemSize;
```

Comment:
This should just use itemcomponent to be honest, 2 x stack items should be twice the size of 1 x stack item.

Reply:
I chose to do it like this so that the size of stacks would be consistent across all instances of an entity. Yes, it does introduce the strange behavior of bulldozing entity sizes, but the test ensures that item sizes are respected by default anyway.